src/Agent.Plugins/PipelineArtifact/PipelineArtifactPlugin.cs

```diff
@@ -36,6 +43,14 @@ public async Task RunAsync(AgentTaskPluginExecutionContext context, Cancellation
             await ProcessCommandInternalAsync(context, targetPath, artifactName, token);
         }
 
+        private string NormalizedJobIdentifier(string jobIdentifier)
+        {
+            Regex rgx = new Regex("[^a-zA-Z0-9 - .]");
```

Comment:
I am not a huge fan of Regex.

Reply:
https://www.quora.com/What-are-the-benefits-of-using-regular-expressions 😄